### ISSUE TITLE
[bug] AuxPow functional test fails intermittently 

### ISSUE THREAD

**Issue by patricklodder** [dogecoin/dogecoin#2406]
# Bug Report

**Current behavior:**

Sometimes, qa/rpc-tests/auxpow.py fails with `not(-11 == 0)`

**Expected behavior**

The test doesn't fail randomly

**Dogecoin Core version**

1.14.0 - 1.14.4-dev

**Error log from CI**

```
auxpow.py:
Initializing test directory /tmp/tests9jm8az_/126
start_node: dogecoind started, waiting for RPC to come up
start_node: RPC successfully started
start_node: dogecoind started, waiting for RPC to come up
start_node: RPC successfully started
Mining blocks...
Stopping nodes

stderr:
Traceback (most recent call last):
  File "/home/runner/work/dogecoin/dogecoin/qa/rpc-tests/auxpow.py", line 88, in <module>
    AuxPOWTest ().main ()
  File "/home/runner/work/dogecoin/dogecoin/qa/rpc-tests/test_framework/test_framework.py", line 164, in main
    stop_nodes(self.nodes)
  File "/home/runner/work/dogecoin/dogecoin/qa/rpc-tests/test_framework/util.py", line 385, in stop_nodes
    stop_node(node, i)
  File "/home/runner/work/dogecoin/dogecoin/qa/rpc-tests/test_framework/util.py", line 380, in stop_node
    assert_equal(return_code, 0)
  File "/home/runner/work/dogecoin/dogecoin/qa/rpc-tests/test_framework/util.py", line 524, in assert_equal
    raise AssertionError("not(%s)" % " == ".join(str(arg) for arg in (thing1, thing2) + args))
AssertionError: not(-11 == 0)

Pass: False, Duration: 2 s
```

**Comment by chromatic** [comment]
I looked into this a little bit and couldn't reproduce. I _think_ what's happening is that one or more of the RPC nodes launched by the test is segfaulting when it's shutting down. Nothing looked obviously wrong in the shutdown code to me.

Perhaps annotating the launch code to cause these nodes to write out core dumps would make this easier to debug?

**Comment by xanimo** [comment]
confimed have experienced this in several of the qa/rpc-tests. point of failure is the `assert_equal(return_code, 0)` within stop_nodes func in qa/rpc-tests/test_framework/util.py, i think L380. still researching the err but at first glance thinking it's to do with timing out:
```
def stop_node(node, i):
    try:
        node.stop()
    except http.client.CannotSendRequest as e:
        print("WARN: Unable to stop node: " + repr(e))
    return_code = bitcoind_processes[i].wait(timeout=BITCOIND_PROC_WAIT_TIMEOUT)
    assert_equal(return_code, 0)
    del bitcoind_processes[i]
```

**Comment by patricklodder** [comment]
Fixed with #2434
